packages/babel-types/src/definitions/experimental.js

```diff
@@ -259,3 +259,17 @@ defineType("StaticBlock", {
   },
   aliases: ["Scopable", "BlockParent"],
 });
+
+// https://github.com/tc39/proposal-js-module-blocks
+defineType("ModuleExpression", {
+  visitor: ["body"],
+  fields: {
+    body: {
```

Comment:
I think we should wrap the module blocks within a `BlockStatement`. Like we did for `DoExpression`: https://github.com/babel/babel/blob/main/packages/babel-parser/ast/spec.md#doexpression

The `BlockStatement` is of `Scopeable` alias, and many more. So when traversing the module body statements in a module block, `path.scope` can be well defined, currently it points to the scope when the `ModuleExpression` defines since it is not `Scopeable`.

Reply:
We could also consider `Program` instead of `BlockStatement`.

Reply:
[`Program`](https://github.com/estree/estree/blob/master/es5.md#programs) allows to include `Directive` in its body. But Module Blocks doesn't seem to allow that.

Reply:
I would be really surprised if module blocks don't allow directives, since they are currently also allowed in blocks and functions.

Reply:
Oh sorry, I misunderstood about directives in ECMAScript spec. Module blocks allows directives. As you said, we can use `Program`.

Reply:
Btw, the original reason I proposed `Program` is that in many transforms we need to inject things to the top-level.
We find the top-level with `path.findParent(p => p.isProgram())`, and this would nicely work out of the box with module blocks.

Reply:
Fixed at e381be0